route/roundrobin.go

```diff
@@ -3,32 +3,44 @@ package route
 import (
 	"context"
 	"log/slog"
+	"math/rand"
+	"sync"
 	"time"
 )
 
 type RoundRobin struct {
 	logger *slog.Logger
 	pool   *EndpointPool
+	lock   *sync.Mutex
+	random *rand.Rand
 
 	initialEndpoint       string
 	mustBeSticky          bool
 	lastEndpoint          *Endpoint
 	locallyOptimistic     bool
 	localAvailabilityZone string
+
+	nextIdx int
 }
 
 func NewRoundRobin(logger *slog.Logger, p *EndpointPool, initial string, mustBeSticky bool, locallyOptimistic bool, localAvailabilityZone string) EndpointIterator {
 	return &RoundRobin{
 		logger:                logger,
 		pool:                  p,
+		lock:                  &sync.Mutex{},
+		random:                rand.New(rand.NewSource(time.Now().UnixNano())),
```

Comment:
Do we use the `ranomd` field somewhere?

Reply:
No. You are right, it should be removed.